src/chainparams.h

```diff
@@ -34,6 +34,10 @@ struct CCheckpointData {
     double fTransactionsPerDay;
 };
 
+static const int DEFAULT_MAINNET_PORT = 8333;
+static const int DEFAULT_TESTNET_PORT = 18333;
+static const int DEFAULT_NOLNET_PORT = 9333;
```

Comment:
whether substitution happens or not is compiler and optimization level dependent.  I'm fine with enums...